cluster/connection_pool.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cluster
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+)
+
+type connectionPool struct {
+	connections map[string]*tlsConn
+	tlsConfig   *tls.Config
+	lock        sync.Mutex
```

Comment:
```suggestion
	mtx        sync.Mutex
```
last nit given you're about to touch this file as part of Chris's comments.

Reply:
I took away the mutex here because the lru cache provides its own locking mechanism, which I think suffices.